chromadb/api/segment.py

```diff
@@ -799,16 +822,64 @@ def get_max_batch_size(self) -> int:
         "SegmentAPI._validate_embedding_record_set", OpenTelemetryGranularity.ALL
     )
     def _validate_embedding_record_set(
-        self, collection: t.Collection, records: List[t.OperationRecord]
+        self,
+        ids: IDs,
+        collection: t.Collection,
+        require_embeddings_or_data: bool,
+        embeddings: Optional[Embeddings] = None,
+        documents: Optional[Documents] = None,
+        uris: Optional[URIs] = None,
+        metadatas: Optional[Metadatas] = None,
     ) -> None:
-        """Validate the dimension of an embedding record before submitting it to the system."""
         add_attributes_to_current_span({"collection_id": str(collection["id"])})
-        for record in records:
-            if record["embedding"]:
-                self._validate_dimension(
-                    collection, len(record["embedding"]), update=True
+
+        validate_batch(
+            (ids, embeddings, metadatas, documents, uris),
+            {"max_batch_size": self.get_max_batch_size()},
+        )
+
+        if (
+            require_embeddings_or_data
+            and embeddings is None
+            and documents is None
+            and uris is None
+        ):
+            raise ValueError("You must provide embeddings, documents, or uris.")
+
+        validate_ids(ids)
+        validate_embeddings(embeddings) if embeddings is not None else None
+        validate_metadatas(metadatas) if metadatas is not None else None
+
+        entities: List[
```

Comment:
this is a bit fragile in changing to external types - i.e if we in the future added a new field, this would not break or update!